integration-tests/tests/test_identity_transformation.py

```diff
@@ -0,0 +1,139 @@
+@text_datasets
+def test_clp_identity_transform(
+    request,
+    package_config: PackageConfig,
+    dataset_logs_fixture: str,
+) -> None:
+    dataset_logs: DatasetLogs = request.getfixturevalue(dataset_logs_fixture)
+    dataset_name = dataset_logs.name
+
+    binary_path_str = str(package_config.clp_bin_dir / "clp")
+    download_dir = package_config.uncompressed_logs_dir / dataset_name
+    archives_dir = package_config.test_output_dir / f"{dataset_name}-archives"
+    extract_dir = package_config.test_output_dir / f"{dataset_name}-logs"
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+    # fmt: off
+    compression_cmd = [
+        binary_path_str,
+        "c",
+        "--progress",
+        "--remove-path-prefix", str(download_dir),
+        str(archives_dir),
+        str(download_dir),
+    ]
+    # fmt: on
+    run_and_assert(compression_cmd)
+
+    extraction_cmd = [binary_path_str, "x", str(archives_dir), str(extract_dir)]
+    run_and_assert(extraction_cmd)
+
+    assert is_dir_tree_content_equal(
+        download_dir, extract_dir
+    ), "Mismatch between clp compression input and decompression output."
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+
+@pytest.mark.clp_s
+@json_datasets
+def test_clp_s_identity_transform(
+    request,
+    package_config: PackageConfig,
+    dataset_logs_fixture: str,
+) -> None:
+    dataset_logs: DaatsetLogs = request.getfixturevalue(dataset_logs_fixture)
+    dataset_name = dataset_logs.name
+
+    binary_path_str = str(package_config.clp_bin_dir / "clp-s")
+    download_dir = package_config.uncompressed_logs_dir / dataset_name
+    archives_dir = package_config.test_output_dir / f"{dataset_name}-archives"
+    extract_dir = package_config.test_output_dir / f"{dataset_name}-logs"
+
+    shutil.rmtree(archives_dir, ignore_errors=True)
+    shutil.rmtree(extract_dir, ignore_errors=True)
+
+    run_and_assert([binary_path_str, "c", str(archives_dir), str(download_dir)])
+    run_and_assert([binary_path_str, "x", str(archives_dir), str(extract_dir)])
+
+    # Recompress the decompressed single-file output and decompress it again to verify consistency.
+    # TODO: Remove this check once we can directly compare decompressed logs (which would preserve
+    #       the directory structure and row/key order) with the original downloaded logs.
+    # See also: https://docs.yscope.com/clp/main/user-guide/core-clp-s.html#current-limitations
+    single_file_archives_dir = (
+        package_config.test_output_dir / f"{dataset_name}-single-file-archives"
+    )
+    single_file_extract_dir = package_config.test_output_dir / f"{dataset_name}-single-file-logs"
+
+    shutil.rmtree(single_file_archives_dir, ignore_errors=True)
+    shutil.rmtree(single_file_extract_dir, ignore_errors=True)
+
+    run_and_assert([binary_path_str, "c", single_file_archives_dir, extract_dir])
```

Comment:
Wonder if you know why we don't need to convert `single_file_archives_dir` to str? (I believe it's type is a path)?

actually if there's a way to make it works, then we can also remove all the explicit conversion in other run_and_assert call

Reply:
I think we should. Thanks for the catch.